src/modules/ipfs/pinata.service.ts

```diff
@@ -39,15 +37,29 @@ export class PinataService {
   }
 
   async uploadText(fileMetadata: FileContent): Promise<UploadToIpfsResult> {
-    const url = `${process.env.PINATA_API_URL}/pinning/pinJSONToIPFS`;
+    const url = `${process.env.PINATA_API_URL}/v3/files`;
+    const jsonString = JSON.stringify(fileMetadata, null, 2);
+
+    // 3. Create a readable stream from the string
+    const jsonStream = Readable.from([jsonString]);
+
+    // 4. Create FormData and append the "file" stream
+    const formData = new FormData();
+    formData.append('file', jsonStream, {
+      filename: 'metadata.json',
+      contentType: 'application/json',
+    });
+
+    formData.append('network', 'public');
     try {
-      const response = await axios.post(url, fileMetadata, {
+      const response = await axios.post(url, formData, {
         headers: {
+          'Content-Type': `multipart/form-data; boundary=${formData._boundary}`,
```

Comment:
Maybe axios should have automatically handled the `formData` and should have properly applied the headers.
 - https://axios-http.com/docs/multipart